eng/init-detect-binaries.proj

```diff
@@ -4,21 +4,33 @@
     <TargetFramework>$(NetCurrent)</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(ToolsDir)BinaryToolKit\BinaryToolKit.csproj" />
+  </ItemGroup>
+
+  <UsingTask TaskName="BinaryToolTask" AssemblyFile="$(BinaryToolKitAssembly)" TaskFactory="TaskHostFactory" />
+
+  <PropertyGroup>
+    <BinariesMode Condition="'$(BinariesMode)' == ''">Validate</BinariesMode>
+    <AllowedBinariesFile Condition="'$(AllowedBinariesFile)' == ''">$(RepoRoot)eng/allowed-vmr-binaries.txt</AllowedBinariesFile>
+  </PropertyGroup>
+
   <Target Name="DetectBinaries"
           BeforeTargets="Build"
           Condition="'$(SkipDetectBinaries)' != 'true'">
-    
-    <PropertyGroup>
-      <DetectBinariesLogLevel Condition="'$(DetectBinariesLogLevel)' == ''">None</DetectBinariesLogLevel>
-      <DetectBinariesAdditionalArgs>--with-sdk $(DotNetRoot)</DetectBinariesAdditionalArgs>
-      <DetectBinariesAdditionalArgs Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(DetectBinariesAdditionalArgs) --with-packages $(PreviouslySourceBuiltPackagesPath)</DetectBinariesAdditionalArgs>
-    </PropertyGroup>
 
-    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Detecting binaries." />
+    <Error
+      Condition="!Exists('$(AllowedBinariesFile)')" 
+      Text="The specified allowed-binaries file does not exist: $(AllowedBinariesFile)" />
 
-    <Exec Command="$(RepositoryEngineeringDir)/detect-binaries.sh --output-dir &quot;$(BinariesReportDir)&quot; --log-level $(DetectBinariesLogLevel) $(DetectBinariesAdditionalArgs)"
-        IgnoreExitCode="true" />
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Detecting binaries." />
```

Comment:
I would avoid timestamps here. CI provides timestamps automatically in logs. Locally, binlogs provide them and also more accurately.